### ISSUE TITLE
How to avoid FloodException?

### ISSUE THREAD

**Issue by Ozymandy** [sochix/TLSharp#407]
Sometimes when I send request I get FloodException. Especially when I try to create channel and set link to it(I mean @name). How to avoid floodexception?

**Comment by barsham** [comment]
You need to use Threading.Sleep and wait for suggested period and retry the last action.

**Comment by Ozymandy** [comment]
@barsham yes, but then you will be banned for a day. So it's not a full solution

**Comment by barsham** [comment]
Actually I had created an exporter application to export messages from Telegram Groups and I used it for exporting more than 5000 messages and I never got banned for a day. But I had to embed Sleep in my engine. I suspect that you might want to do something that alerts the Telegram core security.
Why you are trying to create so many channels?

**Comment by Ozymandy** [comment]
Not many channels. Just 3 channels. And for what time you made wait?

**Comment by barsham** [comment]
```
catch (FloodException floodException)
{
Thread.Sleep(floodException.TimeToWait);
}
```

The maximum was 4 seconds.
